src/docfx/SubCommands/MetadataMerger.cs

```diff
@@ -104,6 +111,103 @@ private static void MergeToc(MetadataMergeParameters parameters, string outputBa
                 YamlMime.TableOfContent);
         }
 
+        private void InitializeTagsRecord(FileModel model)
+        {
+            var content = model.Content as PageViewModel;
+            if (content != null && model.Uids.Length > 0)
+            {
+                _tagsRecord.Add(model.Uids.First().Name, content.Metadata);
+            }
+        }
+
+        private void SetTocViewModelMetadata(TocViewModel vm, ImmutableList<string> metaNames)
+        {
+            foreach (var item in vm)
+            {
+                foreach (var metaName in metaNames)
+                {
+                    MergeMetadata(item, metaName);
+                }
+            }
+        }
+
+        private List<string> MergeMetadata(TocItemViewModel item, string metaName)
+        {
+            Dictionary<string, object> metadata;
+            if (_tagsRecord.TryGetValue(item.Uid, out metadata))
+            {
+                object metaValue;
+                if (metadata.TryGetValue(metaName, out metaValue))
```

Comment:
is there any particular reason? 

---
In reply to: [90403312](https://github.com/dotnet/docfx/pull/941#discussion_r90403312) [](ancestors = 90403312)